### ISSUE TITLE
False positive "Settings updated" in AdminProducts

### ISSUE THREAD

**Issue by madsoliver** [PrestaShop/PrestaShop#19059]
If merchant has a browser window open in AdminProducts, inside a product.
Then he gets logged out (either by clearing cookies, or by login is outdated).
Then logs back into another window, leaving the old product-window open.
Then makes changes and clicks "save" - he gets a positive feedback "Settings updated", when actually in the background the ajax call is redirected to the `security/compromized`-page. 

1. Should the `security/compromized`-page give some other response code than 200?
2. Is there missing a better check in AdminProduct ajax-call since it states "successfull" even when it's not?
3. When this happens, would it be possible to get a popup like this. "You are trying to save, but the token is invalid. [Save and ignore security warning], [Cancel]".
4. Similarely when you get logged out in the background and have worked "for hours" in a browser window, then you get "Could not update settings", and the save button is now disabled (making it impossible to save, even if you log in in another browser window). 

PS 1.7.6.4

**Comment by atomiix** [comment]
Hi @onklmaps 

Thanks for your feedbacks. @PrestaShop/prestashop-product-team, what do you think?

**Comment by colinegin** [comment]
I think this is more a question to @PrestaShop/prestashop-core-developers

**Comment by colinegin** [comment]
@atomiix did you reproduce this bug (it seems it is more a bug than an improvement from what i can read)? If yes, then you can just label it as bug and put it in the backlog, no need for our opinion !

**Comment by LouiseBonnard** [comment]
@atomiix, did you reproduce it? is it a bug?

**Comment by atomiix** [comment]
@LouiseBonnard, yes, I reproduced it and yes it's a bug. I didn't try with 1.7.5 though.

**Comment by atomiix** [comment]
@LouiseBonnard, just tested with 1.7.5.2 and I reproduce the bug!

**Comment by sam-pires** [comment]
@atomiix, can you give us more info about the problem.
And what other improvements these issues highlights so that we can write it down and define cases if needed.

**Comment by colinegin** [comment]
ping @MatShir this is related to the product page rework.
Study is needed to define what should be done.

**Comment by Hlavtox** [comment]
Closed, not relevant anymore.
